app/lib/session_encryptor.rb

```diff
@@ -0,0 +1,63 @@
+class SessionEncryptor
+  KEY_LEN = ActiveSupport::MessageEncryptor.key_len
+
+  def initialize(session_signing_key:)
+    @session_signing_key = session_signing_key
+  end
+
+  def encrypt_session(access_token:, refresh_token:)
+    encrypt_string(
+      {
+        access_token: Base64.encode64(access_token),
+        refresh_token: Base64.encode64(refresh_token),
+      }.to_json,
+    )
+  end
+
+  def decrypt_session(ciphertext)
+    return if ciphertext.blank?
+
+    plaintext = decrypt_string(ciphertext)
+    return unless plaintext
+
+    hash = JSON.parse(plaintext)
```

Comment:
Stupid question, the keys in this are only going to be string rights?
JSON only ever stores strings... unlike YAML... so there's no chance we end up with symbols here that fail on L26/L27?

Reply:
Yeah, it's an object with string keys and string values